book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -47,121 +58,130 @@ In addition to calculations the block subsidy requires constants defined in the
-| ~Second Halving..~                     | ~100%~   | ~0%~           | ~0%~            | ~Yes~             |
+| Height                                 | Miner    | Founder reward | Funding streams | Shielded Coinbase | Target Spacing |
+|----------------------------------------|----------|----------------|-----------------|-------------------|----------------|             
+| ~Genesis~                              | ~**0%**~ | ~**0%**~       | ~**0%**~        | ~**No**~          | ~**None**~     |
+| ~Slow Start Shift..Slow Start Interval~| ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| ~Slow Start Interval..Overwinter~      | ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| ~Overwinter..Sapling~                  | ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| Sapling..Blossom                       | 80%      | 20%            | 0%              | No                | 150 seconds    |
+| Blossom..Heartwood                     | 80%      | 20%            | 0%              | No                | 75 seconds     |
+| Heartwood..Canopy                      | 80%      | 20%            | 0%              | Yes               | 75 seconds     |
+| Canopy..Second Halving                 | 80%      | 0%             | 20%             | Yes               | 75 seconds     |
+| Second Halving..                       | 100%     | 0%             | 0%              | Yes               | 75 seconds     |
 
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
 Given the module structure proposed above all the block subsidy calculations from the protocol must be implemented. The final goal is to do checks for each incoming block and make sure they pass the consensus rules.
 
-To do the calculations and checks the following constants and functions need to be introduced:
+To do the calculations and checks the following constants, types and functions need to be introduced:
 
 ## Constants
 
-- `SLOW_START_INTERVAL`
-- `SLOW_START_SHIFT`
-- `MAX_BLOCK_SUBSIDY`
-- `PRE_BLOSSOM_POW_TARGET_SPACING`
-- `POST_BLOSSOM_POW_TARGET_SPACING`
-- `BLOSSOM_POW_TARGET_SPACING_RATIO`
-- `PRE_BLOSSOM_HALVING_INTERVAL`
-- `POST_BLOSSOM_HALVING_INTERVAL`
-- `FOUNDERS_FRACTION_DIVISOR`
-- `FOUNDER_ADDRESS_CHANGE_INTERVAL`
-- `FOUNDERS_REWARD_ADDRESSES_MAINNET`
-- `FOUNDERS_REWARD_ADDRESSES_TESTNET`
+- `SLOW_START_INTERVAL: Height`
+- `SLOW_START_SHIFT: Height`
+- `MAX_BLOCK_SUBSIDY: u64`
+- `PRE_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `POST_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO: u64`
+- `PRE_BLOSSOM_HALVING_INTERVAL: Height`
+- `POST_BLOSSOM_HALVING_INTERVAL: Height`
+- `FOUNDERS_FRACTION_DIVISOR: u64`
+- `FOUNDER_ADDRESS_CHANGE_INTERVAL: u64`
+- `FOUNDER_ADDRESSES_MAINNET: [&'static str; 48]`
+- `FOUNDER_ADDRESSES_TESTNET: [&'static str; 48]`
```

Comment:
There are 51 testnet addresses:
https://zips.z.cash/zip-0214#testnet-recipient-addresses

Let's make sure we reference ZIP-207, ZIP-214, and ZIP-1014 in this design.

```suggestion
- `FOUNDER_ADDRESSES_TESTNET: [&'static str; 51]`
```

Reply:
This is founders reward, you are referring to funding streams. Founders reward is defined at https://zips.z.cash/protocol/canopy.pdf#foundersreward and it is 48 addresses.

Something missing that we should probably document and implement is a note below the testnet founder reward addresses:

_Note: For Testnet only, the addresses from index 4 onward have been changed from what was implemented at launch. This reflects an upgrade on Testnet , starting from block height 53127. [Zcash-Issue2113]_

Reply:
Ah, right, so we're missing the funding stream address lists in the funding stream section.

Reply:
> Ah, right, so we're missing the funding stream address lists in the funding stream section.

Added them at https://github.com/ZcashFoundation/zebra/pull/1129/commits/5cff70b25f6b5550d475ba85dfe69b9bfe094187

In regards to the note about the change of some founder reward addresses of the testnet at block `53_127`. This is a block before overwinter(Testnet overwinter is at `207_500`) so probably there is no need to document it here but we have to consider it for the checkpoint verifier.

I was checking how zcashd deals with this and i was not able to find the old addresses stored anywhere in the current codebase. The new addresses and the note were added to the protocol at https://github.com/zcash/zips/pull/120 and implemented at https://github.com/zcash/zcash/pull/2114/files but i cant see any logic to deal with them at different heights. Maybe @daira can help.

Reply:
The CheckpointVerifier just checks block hashes. So it doesn't care about transactions or addresses - instead, the checkpoints ensure that we only store pre-Sapling blocks from the best chain.

So I don't think we need to change anything in the CheckpointVerifier or in this design. Pre-Sapling blocks are out of scope for this design.

Reply:
The addresses that were changed were those used after block 53127, I believe. So there was no need to have conditional code; we just required people to upgrade testnet nodes before then.

Reply:
Ok, thank you both, i am going to mark this as resolved in this context.